server/service/src/sync/api_v6/upload_file.rs

```diff
@@ -1,131 +1,76 @@
-};
-use repository::sync_file_reference_row::SyncFileReferenceRow;
+use repository::SyncFileReferenceRow;
+use reqwest::multipart;
 use reqwest::Client;
+use std::fs::File;
 use std::io::Read;
 
 impl SyncApiV6 {
     pub async fn upload_file(
         &self,
-        static_file_service: &StaticFileService,
         sync_file_reference_row: &SyncFileReferenceRow,
-    ) -> Result<(), FileSyncError> {
+        file_name: &str,
+        file_handle: File,
+    ) -> Result<(), SyncApiErrorV6> {
         let Self {
             sync_v5_settings,
             url,
         } = self;
 
-        let route = "upload_file/";
-        let url = url
-            .join(route)
-            .unwrap()
-            .join(&sync_file_reference_row.id)
-            .unwrap(); // Unwrap should be safe here, because we know that sync_file.id is a UUID
+        let route = "upload_file";
+        let url = url.join(route).unwrap(); // Unwrap should be safe here, because we know that sync_file.id is a UUID
 
-        // Get file path
-        let file = static_file_service
-            .find_file(
-                &sync_file_reference_row.id,
-                StaticFileCategory::SyncFile(
-                    sync_file_reference_row.table_name.to_owned(),
-                    sync_file_reference_row.record_id.to_owned(),
-                ),
-            )
-            .map_err(|err| {
-                log::error!("Error from static_file_service: {:#?}", err);
-                FileSyncError::CantFindFile("Error from static_file_service".to_string())
-            })?;
-        let file = match file {
-            Some(file) => file,
-            None => {
-                return Err(FileSyncError::CantFindFile(
-                    "File doesn't exist in static_file_service".to_string(),
-                ))
+        let error_with_url = |source: SyncApiErrorVariantV6| -> SyncApiErrorV6 {
+            SyncApiErrorV6 {
+                url: url.clone(),
+                route: route.to_string(),
+                source,
             }
         };
 
-        let mut file_handle = std::fs::File::open(file.path.clone()).map_err(|err| {
-            log::error!("Error opening file: {:#?}", err);
-            FileSyncError::StdIoError(err)
-        })?;
-
-        // Read the file into memory (ideally could be a stream or something, and upload the file in chunk so we can stop quickly when sync starts/stops/pauses)
-        let mut file_bytes = Vec::new();
-        file_handle.read_to_end(&mut file_bytes).map_err(|err| {
-            log::error!("Error reading file: {:#?}", err);
-            FileSyncError::StdIoError(err)
-        })?;
+        let client = Client::new();
 
-        // This is an example request structure, it just exists to give a compiler warning if the structure changes
-        // You'll need to modify the actual request using the reqwest::multipart::Parts below if there's a change here...
-        let _canary_request = SyncUploadFileRequestV6 {
```

Comment:
_canary_request removed. There is multipart to request method below with comments about mapping